dace/transformation/estimator/soap/einsum_to_sdfg.py

```diff
@@ -0,0 +1,175 @@
+            A = np.random.rand(inp_dim**order).reshape([inp_dim] * order)           
+            arrays.append(A)
+            
+
+    # Extract symbols
+    path_info = oe.contract_path(subscripts, *arrays)
+
+    counter = 0
+    array_names = []
+    unique_arrays = []
+    dace_arrays = {}
+    symbols = {}
+    symidx = 0
+    array_shapes = {}
+    tokens = re.split(',|->', subscripts)
+    for arr, t in zip(arrays, tokens[:-1]):
+        try:
+            idx = unique_arrays.index(arr)
+            arr_name = f'inp{idx}'
+        except ValueError:
+            unique_arrays.append(arr)
+            arr_name = f'inp{counter}'
+            counter += 1
+            dace_arrays[arr_name] = arr
+            shape = []
+            for char in t.strip():
+                if char not in symbols:
+                    newsym = dace.symbol(f'S{symidx}')
+                    symidx += 1
+                    symbols[char] = newsym
+                shape.append(symbols[char])
+            array_shapes[arr_name] = shape
+        array_names.append(arr_name)
+    
+    sdfg = dace.SDFG('tensor_contractions')
+    for name, arr in dace_arrays.items():
+        sdfg.add_array(name, array_shapes[name], dt[arr.dtype.type])
+    
+    counter = 0
+    state = None
+    for contraction in path_info[1].contraction_list:
+
+        print(contraction)
+
+        tokens = re.split(',|->', contraction[2])
+        assert(len(tokens) == 3)
+
+        first_idx, second_idx = contraction[0]
+        if first_idx > second_idx:
+            first_inp = array_names.pop(first_idx)
+            second_inp = array_names.pop(second_idx)
+        else:
+            second_inp = array_names.pop(second_idx)
+            first_inp = array_names.pop(first_idx)
+
+        first_arr = dace_arrays[first_inp]
+        second_arr = dace_arrays[second_inp]
+        output = np.einsum(contraction[2], first_arr, second_arr)
+        out_name = f'out{counter}'
+        counter += 1
+        array_names.append(out_name)
+        dace_arrays[out_name] = output
+        output_shape = [symbols[char] for char in tokens[-1].strip()]
+        sdfg.add_transient(out_name, output_shape, dt[output.dtype.type])
+
+        prv_state = state
+        state = sdfg.add_state(f'contraction_{first_inp}_{second_inp}')
+        if prv_state:
+            sdfg.add_edge(prv_state, state, dace.InterstateEdge())
+
+        map_ranges = {}
+        needs_wcr = set(tokens[0]).union(set(tokens[1])) != set(tokens[2])
+        wcr = None
+        if needs_wcr:
+            wcr = 'lambda a, b: a + b'
+        for t, arr in zip(tokens, [first_arr, second_arr, output]):
+            for i, idx in enumerate(t.strip()):
+                if idx in map_ranges.keys():
+                    continue
+                # map_ranges[idx] = (0, arr.shape[i]-1, 1)
+                map_ranges[idx] = (0, symbols[idx]-1, 1)
+        in_memlets = {}
+        in_memlets['__in0'] = dace.Memlet(
+                data=first_inp,

```

Comment:
We should probably switch to `dace.Memlet.simple` or `dace.Memlet(expr)` here since the `data` parameter has been deprecated. @tbennun what is the official (and efficient) way to create Memlets now?